Library/Formula/opencoarrays.rb

```diff
@@ -0,0 +1,67 @@
+class Opencoarrays < Formula
+  desc "open-source coarray Fortran ABI, API, and compiler wrapper"
+  homepage "http://opencoarrays.org"
+  url "https://github.com/sourceryinstitute/opencoarrays/releases/download/1.3.2/OpenCoarrays-1.3.2.tar.gz"
+  sha256 "391937a5d24962bd93ecec2402545a3e9d6c7a43987082c7de7dee3b74151ec3"
+
+  head "https://github.com/sourceryinstitute/opencoarrays.git"
+
+  option "without-test", "Skip build time tests (requires properly configured MPICH and firewall settings)"
+
+  depends_on :fortran
+  depends_on :mpi => [:cc, :f90]
+  depends_on "cmake" => :build
+
+  # GCC >= 5.2 required; no other Homebrew supported compilers are compatible
+  # but the mpicc compiler may be clang (which is ABI compatible with GCC)
+  fails_with :llvm
+  fails_with :gcc
+  fails_with :gcc_4_0
+  ("4.3".."4.9").each do |series|
+    fails_with :gcc => series
+  end
+  fails_with :gcc => "5.1"
+
+  def install
+    mkdir "build" do
+      ENV["FC"] = ENV["MPIFC"]
+      ENV["CC"] = ENV["MPICC"]
+      cmake_args = std_cmake_args
+      system "cmake", "..", *cmake_args
+      system "make"
+      # A bug in GFortran <= 5.2 causes the "co_reduce" test to fail; find
```

Comment:
@mikemcquaid OK, I'll ack for `JavaRequirement` in the Formula subdir... Just FYI, my ruby is limited to my experience contributing to Homebrew, so I may ping you (or maybe someone else) if I get stuck.... Any links you can provide to relavent source code would be *highly* appreciated. :bow: